backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt

```diff
@@ -104,4 +104,111 @@ class ShorkInterpreterControllerV1IT() : AbstractControllerTest() {
         assertEquals("FINISHED", responseData["gameState"])
         assertEquals("B", responseData["result.winner"])
     }
+
+    @Test
+    fun `test player code submission with valid username`() = runTest {
+        val player = "playerA"
+        client.post("/api/v1/lobby/0/code/$player") {
+            contentType(ContentType.Application.Json)
+            setBody("someString")
+        }
+        val result = client.get("/api/v1/lobby/0/code/playerA")
+        assertEquals(
+            "someString",
+            Json.parseToJsonElement(result.bodyAsText()).jsonObject["code"]?.jsonPrimitive?.content,
+        )
+    }
+
+    @Test
+    fun `test  player valid code submission`() = runTest {
+        val player = "playerA"
+        val result = client.post("/api/v1/lobby/0/code/$player")
+        assertEquals(HttpStatusCode.OK, result.status)
+    }
+
+    @Test
+    fun `test player code submission with invalid username`() = runTest {
+        val player = "playerC"
+        client.post("/api/v1/lobby/0/code/$player") {
+            contentType(ContentType.Application.Json)
+            setBody("someString")
+        }
+        val result = client.get("/api/v1/lobby/0/code/playerC")
+        assertEquals(HttpStatusCode.BadRequest, result.status)
+        assert(result.bodyAsText().contains("No player with that name in the lobby"))
+    }
+
+    @Test
+    fun `test player code submission reflecting in lobby status`() = runTest {
+        val player = "playerA"
+        client.post("/api/v1/lobby/0/code/$player") {
+            contentType(ContentType.Application.Json)
+            setBody("someString")
+        }
+
+        val result = client.get("/api/v1/lobby/status/0")
+        val responseData = parseStatus(result)
+        assertEquals("true", responseData["playerASubmitted"])
+        assertEquals("false", responseData["playerBSubmitted"])
+        assertEquals("NOT_STARTED", responseData["gameState"])
+        assertEquals("UNDECIDED", responseData["result.winner"])
+    }
+
+    @Test
+    fun `test code submission by invalid player username reflects no change`() = runTest {
```

Comment:
i'd insert a "status" in the test name here
`test code submission with invalid username does not change status`
or something similar